src/idl_gen_kotlin.cpp

```diff
@@ -159,6 +159,24 @@ class KotlinGenerator : public BaseGenerator {
     }
   }
 
+  // with the addition of optional scalar types,
+  // we are adding the nullable '?' operator to return type of a field.
+  std::string GetterReturnType(const FieldDef &field) const {
+    auto base_type = field.value.type.base_type;
+
+    auto r_type = GenTypeGet(field.value.type);
+    if (field.IsScalarOptional() ||
```

Comment:
Do you not need to check for required fields?

I'm hoping field.optional is enough to determine whether the output field is optional or not - (though we're already inconsistent around vectors as nonpresense is null in C++/Rust but empty-vector in swift)

Reply:
>Do you not need to check for required fields?

Scalar fields can't have `required` attribute.
@aardappel Merge it?